lib/permission.js

```diff
@@ -17,21 +23,164 @@ module.exports = api;
 
 var logger = bedrock.loggers.get('app');
 
-bedrock.events.on('bedrock.start', init);
+bedrock.events.on('bedrock-mongodb.ready', function(callback) {
+  PERMISSIONS = bedrock.config.permission.permissions;
+  async.auto({
+    openCollections: function(callback) {
+      database.openCollections(['roles'], callback);
+    },
+    createIndexes: ['openCollections', function(callback) {
+      database.createIndexes([{
+        collection: 'roles',
+        fields: {id: 1},
+        options: {unique: true, background: false}
+      }], callback);
+    }],
+    addRoles: ['createIndexes', function(callback) {
+      async.eachSeries(
+        bedrock.config.permission.roles, function(role, callback) {
+        role.permanent = true;
+        logger.debug('adding role', role);
+        api.addRole(null, role, function(err) {
+          if(err && database.isDuplicateError(err)) {
+            err = null;
+          }
+          callback(err);
+        });
+      }, callback);
+    }]
+  }, callback);
+});
 
-function init(callback) {
-  // log permissions
-  var permissions = bedrock.config.permission.permissions;
-  Object.keys(permissions).forEach(function(permission) {
-    logger.debug('adding permission', permissions[permission]);
-  });
```

Comment:
We still want to log the addition of permissions to the system. Perhaps move this up into a `bedrock.init` section?